agasc/supplement/magnitudes/mag_estimate_report.py

```diff
@@ -27,6 +28,48 @@
 logger = logging.getLogger('agasc.supplement')
 
 
+class TableEncoder(json.JSONEncoder):
+    """
+    Utility class to encode tables as json.
+
+    Example::
+
+        >>> import json
+        >>> from astropy.table import Table, Column, MaskedColumn
+        >>> a = MaskedColumn([1, 2], name='a', mask=[False, True], dtype='i4')
+        >>> b = Column([3, 4], name='b', dtype='i8')
+        >>> t = Table([a, b])
+        >>> print(json.dumps(t, cls=TableEncoder, indent=2))
+        {
+          "a": [
+            1,
+            999999
+          ],
+          "b": [
+            3,
+            4
+          ]
+        }
+    """
+    def default(self, obj):
+        if isinstance(obj, table.Table):
+            return {k: obj[k] for k in obj.colnames}
+        if isinstance(obj, table.MaskedColumn):
+            fv = obj.get_fill_value()
+            return [(fv if mask else val) for mask, val in zip(obj.mask, obj)]
+        if isinstance(obj, table.Column):
+            return [val for val in obj]
+        if isinstance(obj, table.Row):
+            return obj.values()
+        if isinstance(obj, np.ma.core.MaskedConstant):
+            return obj.view().fill_value
+        if np.isscalar(obj):
+            return int(obj)
+        if np.isreal(obj):
+            return float(obj)
```

Comment:
I think this can be replaced with the one-liner:
```suggestion
        if isinstance(obj, table.Table):
            return {name: val.tolist() for name, val in obj.columns.items()}
```
This puts `None` in the output for masked values, which is fine because JSON serializes that as `null`. That captures the intent of the missing value more than using the fill value which is not easy for downstream applications to handle (meaning to get back to the original masked column).

The only thing this version does not do is serialize a Table Row, so that would be separate. I'm not sure why that was included since it seems out of scope from the docstring "encode tables as json".

Reply:
I added the row conversion because I needed it. If anything, the docstring is outdated. I would have to go back and check where exactly a row was serialized.

Reply:
In the end, I followed this suggestion.